packages/engine/lib/experiment-control/src/environment/logging.rs

```diff
@@ -258,6 +258,20 @@ pub fn init_logger<P: AsRef<Path>>(
         ),
     };
 
+    if !Path::new(log_folder).exists() {
```

Comment:
I think `log_folder` already is a `Path`
```suggestion
    if !log_folder.exists() {
```